curated_transformers/repository/fsspec.py

```diff
@@ -0,0 +1,108 @@
+from dataclasses import dataclass
+from typing import IO, Any, Dict, Optional
+
+from fsspec import AbstractFileSystem
+
+from .repository import Repository, RepositoryFile
+
+
+@dataclass
+class FsspecArgs:
+    """
+    Convenience wrapper for additional fsspec arguments.
+    """
+
+    kwargs: Dict[str, Any]
+
+    def __init__(self, **kwargs):
+        """
+        Keyword arguments are passed through to the fsspec implementation.
+        Construction may raise in the future when reserved arguments like
+        ``mode`` or ``encoding`` are used.
+        """
+        # Future improvement: raise on args that are used by fsspec methods,
+        # e.g. `mode` or `encoding`.
+        self.kwargs = kwargs
+
+
+class FsspecFile(RepositoryFile):
+    """
+    Repository file on an fsspec filesystem.
```

Comment:
Could we also add a link to the `fsspec` docs here (like we do below)?

Reply:
Done